### PR TITLE
Home 4

### PR DESCRIPTION
**Pull request by YKotov**

### CODE REVIEW

File: pet-app/src/app/pages/courses/data-sharing.service.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+@Injectable()
+export class DataSharingService {
+
+  private searchedElement = new BehaviorSubject('');
+  getSearchedElement = this.searchedElement.asObservable();
+
+  constructor() {
+  }
+
+  serchedString(filtered: string) {
```
Comment (torin4fan) [inline pet-app/src/app/pages/courses/data-sharing.service.ts]: syntax error -> seArchedString

File: pet-app/src/app/pages/courses/search/search.component.ts [hunk 2]
```diff
@@ -11,17 +12,20 @@ import {
...
   logging() {
-    this.consoledElement.log(this.consoledElement.data);
-    this.consoledElement.data = null ? 'empty' : '';
+    console.log(this.cFilter);
   }
 
-  constructor() {
-    this.consoledElement = console;
-    this.logging();
+  constructor(private data: DataSharingService) {
```
Comment (torin4fan) [inline pet-app/src/app/pages/courses/search/search.component.ts]: more predictable would be use naming as
private dataSharingService: DataSharingService

File: pet-app/src/app/pages/courses/search/search.component.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <div class="actions-wrapp">
   <div class="search-wrapp">
-    <input type="text" [(ngModel)]="consoledElement.data"  placeholder="Name, topic or date" class="search">
+    <input type="text" [(ngModel)]="cFilter" (keyup)="search()"  placeholder="Name, topic or date" class="search">
```
Comment (torin4fan) [inline pet-app/src/app/pages/courses/search/search.component.html]: Filter courses list by name when user enters some text in the field ‘search’ and presses
the ‘Find’ button.
Use 'button' please

File: pet-app/src/app/pages/courses/search/search.component.ts [hunk 4]
```diff
@@ -11,17 +12,20 @@ import {
...
   }
 
   ngOnInit() {
+    this.data.getSearchedElement.subscribe(cFilter => this.cFilter = cFilter);
```
Comment (torin4fan) [inline pet-app/src/app/pages/courses/search/search.component.ts]: what this code doing?

File: pet-app/src/app/pages/courses/course-list/course-list.component.html [hunk 5]
```diff
@@ -1,3 +1,3 @@
-<div *ngFor="let item of courseItemArr | async" id="{{item._id}}">
-  <app-crs-detail [item]=item (consoledElement)=loggerEvent($event)></app-crs-detail>
+<div *ngFor="let item of courseItemArr | async | filter:cFilter; let i=index"   id="{{item._id}}">
+  <app-crs-detail [item]=item (consoledElement)=loggerEvent($event) *ngIf="i<maxPerPage"></app-crs-detail>
```
Comment (torin4fan) [inline pet-app/src/app/pages/courses/course-list/course-list.component.html]: Do we have requirements for this?
*ngIf="i<maxPerPage"

File: pet-app/src/app/pages/courses/loadmore/loadmore.component.html [hunk 6]
```diff
@@ -1,3 +1,4 @@
 <div>
-  <button class="load-more">Load more</button>
+  <button *ngIf="elements" class="load-more">Load more</button>
+  <button *ngIf="!elements" class="load-more">No data, feel free to add new course</button>
```
Comment (torin4fan) [inline pet-app/src/app/pages/courses/loadmore/loadmore.component.html]: it works strange. Move this message "without button" in course-list.component
Hide 'load more' button if we don't have any courses

File: pet-app/src/app/pages/courses/course-list/crs-detail/crs-detail.component.html [hunk 7]
```diff
@@ -1,14 +1,20 @@
-<div class="content-item">
+<div class="content-item" [appActualizationCrs]="item.data">
   <div class="left-part">
     <div class="head-part">
-      <h3 *ngIf="item.index == 0"> Course entry</h3>
-      <h3 *ngIf="item.index > 0"> Course {{item.index}}</h3>
+      <h3 *ngIf="item.index == 0"> Course entry |
```
Comment (torin4fan) [inline pet-app/src/app/pages/courses/course-list/crs-detail/crs-detail.component.html]: What 'index' is?
Just show/hide 'star' if course rated

File: pet-app/src/app/pages/courses/course-list/course-list.component.html [hunk 8]
```diff
@@ -1,3 +1,3 @@
-<div *ngFor="let item of courseItemArr | async" id="{{item._id}}">
-  <app-crs-detail [item]=item (consoledElement)=loggerEvent($event)></app-crs-detail>
+<div *ngFor="let item of courseItemArr | async | filter:cFilter; let i=index"   id="{{item._id}}">
```
Comment (torin4fan) [inline pet-app/src/app/pages/courses/course-list/course-list.component.html]: what id="{{item._id}}" give us?

Comment (YKotov) [reply pet-app/src/app/pages/courses/course-list/course-list.component.html]: http://prntscr.com/kxo2cq

**Pull request merged**